fe/fe-core/src/main/java/com/starrocks/catalog/JDBCTable.java

```diff
@@ -0,0 +1,131 @@
+    private Map<String, String> jdbcProperties = Maps.newHashMap();
+
+    public JDBCTable() {
+        super(TableType.JDBC);
+    }
+
+    public JDBCTable(long id, String name, List<Column> schema, Map<String, String> properties) throws DdlException {
+        super(id, name, TableType.JDBC, schema);
+        validate(properties);
+    }
+
+    public String getResourceName() {
+        return resourceName;
+    }
+
+    public String getJdbcDatabase() {
+        return jdbcDatabase;
+    }
+
+    public String getJdbcTable() {
+        return jdbcTable;
+    }
+
+    public Map<String, String> getJdbcProperties() {
+        return jdbcProperties;
+    }
+
+    private void validate(Map<String, String> properties) throws DdlException {
+        if (properties == null) {
+            throw new DdlException("Please set properties of jdbc table, they are: jdbc.database, jdbc.table and jdbc.resource");
+        }
+
+        Map<String, String> copiedProps = Maps.newHashMap(properties);
+        jdbcDatabase = copiedProps.get(DB);
+        if (Strings.isNullOrEmpty(jdbcDatabase)) {
+            throw new DdlException(String.format(PROPERTY_MISSING_MSG, DB, DB));
+        }
+        copiedProps.remove(DB);
+
+        jdbcTable = copiedProps.get(TABLE);
+        if (Strings.isNullOrEmpty(jdbcTable)) {
+            throw new DdlException(String.format(PROPERTY_MISSING_MSG, TABLE, TABLE));
+        }
+        copiedProps.remove(TABLE);
+
+        resourceName = copiedProps.get(RESOURCE);
+        if (Strings.isNullOrEmpty(resourceName)) {
+            throw new DdlException("property " + RESOURCE + " must be set");
+        }
+        copiedProps.remove(RESOURCE);
+
+        Resource resource = Catalog.getCurrentCatalog().getResourceMgr().getResource(resourceName);
+        if (resource == null) {
+            throw new DdlException("jdbc resource [" + resourceName + "] not exists");
+        }
+        if (resource.getType() != ResourceType.JDBC) {
+            throw new DdlException("resource [" + resourceName + "] is not jdbc resource");
+        }
+        jdbcProperties = copiedProps;
+
+    }
+
+    // @TODO implement toThrift function
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+        super.write(out);
+
+        JsonObject obj = new JsonObject();
+        obj.addProperty(DB, jdbcDatabase);
+        obj.addProperty(TABLE, jdbcTable);
+        obj.addProperty(RESOURCE, resourceName);
+        for (Map.Entry<String, String> entry : jdbcProperties.entrySet()) {
+            obj.addProperty(entry.getKey(), entry.getValue());
+        }
+        Text.writeString(out, obj.toString());
+    }
+
+    public void readFields(DataInput in) throws IOException {
+        super.readFields(in);
+        String jsonStr = Text.readString(in);
+        JsonObject obj = JsonParser.parseString(jsonStr).getAsJsonObject();
+        jdbcDatabase = obj.getAsJsonPrimitive(DB).getAsString();
+        obj.remove(DB);
```

Comment:
Why remove?

Reply:
@mofeiatwork Avoid to put them into `jdbcProperties`